docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -735,6 +735,22 @@ The output of the query has the following columns:
     - ``array(integer)``
     - The set of field IDs used for equality comparison in equality delete files
 
+``$all_files`` table
+^^^^^^^^^^^^^^^^^^^^
+
+The ``$all_files`` table provides a detailed overview of the data file in all snapshots of the Iceberg table.
+The output of this system table can contain duplicate entries, as corresponding snapshots can refer to the same data files.
+This table is handy to debug the data issues, where the snapshot is rolled back/ bring back to the future from the Trino/Spark
+and now trying to understand what all data-files are present, and is there any implication because of the optimization/rollback operations.
+
+To retrieve the information about the data files from all the snapshots of the Iceberg table ``test_table`` use the following query::
+
+    SELECT * FROM "test_table$all_files"
+
+
+
+The output of the above query has the columns similar to the files-table system table.
+
```

Comment:
With the risk of repeating ourselves, please do add a concrete example of the output for this query (maybe even use a result containing duplicated data files) in line with the explanation you have provided in the paragraph above.

Also do add a description for the table structure.

Reply:
Did not added initially, as was not sure whether we are looking for some dupes or not, so kept it open :)

Reply:
updated @mosabua